src/code/matcher/matcher.cpp

```diff
@@ -69,13 +69,48 @@ const IServerConf *matcher::requestToServer(const IConf *conf, const IListen *so
 const ILocation *matcher::requestToLocation(const IServerConf *s, const BasicHTTPRequest &req) throw()
 {
 	std::list<const ILocation *> list = s->getLocations();
+	std::list<const ILocation *>::const_iterator tmp = list.end();
 	std::list<const ILocation *>::const_iterator it = list.begin();
-
 	while (it != list.end())
 	{
-		if ((*it)->getPath().get() == req.getPath())
-			return *it;
-		++it;
+		if ((req.getPath().find((*it)->getPath().get()) == 0) || (req.getPath() == "" && (*it)->getPath().get() == "/"))
```

Comment:
i dont understand the logic here.
these 2 "if"s are very confusing.
what are you trying to change? why `if ((*it)->getPath().get() == req.getPath())` does not work?